### PR TITLE
GroupBy with baseline hash table modification

### PR DESCRIPTION
**Pull request by bagrorg**

### CODE REVIEW

**Comment by bagrorg** [comment]
@kurapov-peter

File: common/common.hpp [hunk 1]
```diff
@@ -28,11 +28,12 @@ void dump_collection(const Collection &c, std::ostream &os = std::cout) {
 namespace helpers {
 std::vector<uint32_t> make_unique_random(size_t size);
 
-template <class T> std::vector<T> make_random(size_t size) {
+template <class T>
+std::vector<T> make_random(size_t size, size_t l = 1, size_t r = 10000) {
```
Comment (kurapov-peter) [inline common/common.hpp:32]: would benefit from better parameters naming

File: common/dpcpp/hashtable.hpp [hunk 2]
```diff
@@ -87,4 +87,66 @@ template <class Key, class T, class Hash> class SimpleNonOwningHashTable {
...
+private:
+  sycl::global_ptr<Key> _keys;
+  sycl::global_ptr<T> _vals;
+  size_t _size;
+  Hash _hasher;
+  Key _empty_key;
+
+  static constexpr uint32_t elem_sz = CHAR_BIT * sizeof(uint32_t);
+
+  bool update_bitmask_group_by(Key key, T val) {
```
Comment (kurapov-peter) [inline common/dpcpp/hashtable.hpp:134]: Naming is now confusing as you've changed the logic

File: common/dpcpp/hashtable.hpp [hunk 3]
```diff
@@ -87,4 +87,66 @@ template <class Key, class T, class Hash> class SimpleNonOwningHashTable {
       }
     }
   }
+};
+
+template <class Key, class T, class Hash>
+class SimpleNonOwningHashTableForGroupBy {
```
Comment (kurapov-peter) [inline common/dpcpp/hashtable.hpp:93]: Can we just replace the old implementation?

File: tests/hash_table_tests.cpp [hunk 4]
```diff
@@ -191,7 +191,7 @@ TEST(HashTable, BigBuild) {
   std::vector<uint32_t> host_src(host_src_tmp.begin(), host_src_tmp.end());
 
   StaticSimpleHasher<buf_size> hasher;
-  size_t bitmask_sz = (buf_size / 32) ? (buf_size / 32) : 1;
+  size_t bitmask_sz = buf_size / 32 + 1;
```
Comment (kurapov-peter) [inline tests/hash_table_tests.cpp:194]: not a bitmask anymore, isn't it?

**Review by kurapov-peter — CHANGES_REQUESTED** [review]

**Review by kurapov-peter — APPROVED** [review]
LGTM

File: tests/hash_table_tests.cpp (hunk 4 above)
Comment (bagrorg) [reply tests/hash_table_tests.cpp:194]: I'm not sure I understand what needs to be done, and whether it is necessary if we do not replace the old version of the hash table

**Pull request merged**